### PR TITLE
add PR template

### PR DESCRIPTION
**Pull request by kate-goldenring**
Add a pull request template. Also includes notes about reading our contributing guide before submitting a PR and how to link issues.

### CODE REVIEW

**Comment by kate-goldenring** [comment]
Should we add tips on how to use the [SAME VERSION] [IGNORE INTERMEDIATE BUILDS] AND [ALLOW INTERMEDIATE BUILDS] labels?

**Comment by jiria** [comment]
I would say yes, I do not recall seeing this being used with other projects.

**Comment by bfjelds** [comment]
> Should we add tips on how to use the [SAME VERSION] [IGNORE INTERMEDIATE BUILDS] AND [ALLOW INTERMEDIATE BUILDS] flags in the PR title?

my gut would be to explain [SAME VERSION] in the PR template, but explain the others in the contributing md (they aren't likely to be frequently encountered).  but i'm good with explanations for all.

**Review by bfjelds — APPROVED** [review]

File: docs/contributing.md [hunk 1]
```diff
@@ -29,6 +29,16 @@ We follow the [SymVer](https://semver.org/) versioning strategy: [MAJOR].[MINOR]
 
 To ensure that all product versioning is consistent, our CI builds will execute `version.sh -c` to check all known instances of version in our YAML, TOML, and code.  This will also check to make sure that version.txt has been changed.  If a pull request is needed where the version should not be changed, include `[SAME VERSION]` in the pull request title.
 
+## PR title flags
+Akri's workflows check for three flags in the titles of PRs in order to decide whether to execute certain checks. 
+
+The [version check workflow](../.github/workflows/check-versioning.yml) will run, ensuring you have increased the version number, unless you (A) only change a a file that is on an ignored path of the workflow, such as all `*.md` files OR (B) specify the `[SAME VERSION]` flag. Use this flag if your change will trigger the workflow and the version should not be changed due to your PR. The flag will cause the check to automatically succeed.
```
Comment (kate-goldenring) [inline docs/contributing.md:35]: "due to" -> "by"

**Review by bfjelds — APPROVED** [review]

**Pull request merged**